digdag-server/src/main/java/io/digdag/server/metrics/DigdagTimedMethodInterceptor.java

```diff
@@ -0,0 +1,85 @@
+package io.digdag.server.metrics;
+
+import com.google.inject.Inject;
+import io.digdag.metrics.DigdagTimed;
+import io.digdag.spi.TaskRequest;
+import io.digdag.spi.metrics.DigdagMetrics;
+import static io.digdag.spi.metrics.DigdagMetrics.Category;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * MethodInterceptor for Timed annotation.
+ */
+public class DigdagTimedMethodInterceptor implements MethodInterceptor
+{
+    private static Logger logger = LoggerFactory.getLogger(DigdagTimedMethodInterceptor.class);
+    @Inject
+    DigdagMetrics metrics;
+
+    public DigdagTimedMethodInterceptor() { }
+
+    public Object invoke(MethodInvocation invocation)
+            throws Throwable
+    {
+        DigdagTimed timed = invocation.getMethod().getAnnotation(DigdagTimed.class);
+
+        Category category = Category.fromString(timed.category());
+        String value = mkValue(timed, invocation);
+        String metricsName = metrics.mkMetricsName(category, value);
+        Timer.Sample sample = metrics.timerStart(category);
+        try {
+            return invocation.proceed();
+        } catch (Exception ex) {
+            throw ex;
+        } finally {
+            try {
+                Tags taskTags = timed.taskRequest() ? getTaskRequest(invocation) : Tags.empty();
```

Comment:
This method call could be invoked before timerStart call for elapsed time accuracy?

Reply:
Sorry, I may misunderstand your question. 
But timerStart is always call in L.35 as follows.

```
Timer.Sample sample = metrics.timerStart(category);
        try {
            return invocation.proceed();
```
So I guess the L.42 method call is never invoked before timerStart.

Reply:
Sorry, my comment made your confusing. `getTaskRequest(invocation)` method call looks like heavy because it has for loop and instanceof operator. If the method could be called before `timerStart` method call, the metrics could be more accurate. But not so big deal I think.

Reply:
Thanks! Got it. I will fix it before merge.